### DISCUSSION TITLE
TinaCMS escaping text formatting related to links & italics

### DISCUSSION THREAD

**Discussion by emilyrbowe** [tinacms/tinacms#4123] [Help]
Hi, I'm finding that our CMS users are running into issues with text formatting that will get escaped upon save. The particular issue is when something italicized is hyperlinked — the site shows the hyperlink, but without the italics. 

We migrated from Forestry earlier this year and have struggled with many of the shortcodes we had in place there and have found that non-technical users get frustrated easily with TinaCMS because their changes are not saved as they made them in the editor. I, as the admin, have to edit the actual markdown from a code editor and re-commit the file in order for final formatting to stick. 

Any help would be appreciated.

**Comment by kldavis4** [comment]
@jamespohalloran any thoughts on this?

**Comment by jamespohalloran** [comment]
@emilyrbowe Could you provide a video of some rich-text that gets escaped? I'm currently unable to reproduce 
(Also, can you specify which version of tinacms & @tinacms/cli you're on?)
Thanks!

**Reply by emilyrbowe** [reply]
Yep, I will do this later today and get it up here! Thanks for taking the time to check in about it.

**Reply by emilyrbowe** [reply]
Versions details from the package.json are here, @jamespohalloran: 
"@tinacms/cli": "^1.5.5",
"tinacms": "^1.4.5"

**Comment by emilyrbowe** [comment]
Okay, this is happening way later than I had originally planned, but here are some details that just happened to illustrate a related issue. A colleague was trying to change some text that was inside of an "alert box" that we'd set up as a custom shortcode called `alert-box`. She was doing this from inside the Tina CMS editor and when she hit save, all shortcodes on the page had a `\` character added immediately after the opening curly brackets, which breaks the shortcodes. The full commit is linked [here](https://github.com/bplmaps/lmec-main-site/commit/6fe0dcb4dfa193bec3238466623ba0f7d160623c) and a visual snippet of the `alert-box` change is shown below. There are a few instances of things that are hyperlinked and italicized ending up hyperlinked, but with a `*` around the text. 

![image](https://github.com/tinacms/tinacms/assets/7975988/9b13ff6c-2e52-4e42-8455-19f533db5ce1)

![image](https://github.com/tinacms/tinacms/assets/7975988/8834a505-1e06-4fc8-bd2a-6f2b9196a41c)

**Comment by emilyrbowe** [comment]
Just had a thought and wondering if it makes any sense to @jamespohalloran and @kldavis4: Right now, we have these custom Hugo shortcodes (like `alert-box`) in the above screenshot that are inside the rich text editor. I'm wondering if what is happening is that graphql is escaping characters (like the `<` in the shortcode identifier) when it saves to the tina db. I've seen references to graphql doing similar things, so I guess I'm wondering if there if either of you has any idea about how to continue to use hugo custom shortcodes in text areas that are accessible from the cms side?

**Comment by mustafasahin0** [comment]
Just wondering if there is any update on this escape character issue.

**Reply by emilyrbowe** [reply]
Not to my knowledge, @mustafasahin0, but we are still seeing the issues on our end.

**Comment by snotrman** [comment]
So how did you all solve this?

**Reply by emilyrbowe** [reply]
Never did -- we are migrating off Tina as a result :/

**Reply by snotrman** [reply]
This is what I did:
```
  schema: {
    collections: [
      {
        name: "post",
        label: "Posts",
        path: "content/posts",
        match: {
          exclude: "_index.md", // Ensures `_index.md` is not included
        },
        fields: [
          { type: "string", name: "title", label: "Title", isTitle: true, required: true },
          { type: "rich-text", name: "body", label: "Body", isBody: true, parser: { type: "markdown", skipEscaping: "all" } },
        ],
      },
 ```
 
 ```
 parser: { type: "markdown", skipEscaping: "all" } },
 ```

**Reply by emilyrbowe** [reply]
Interesting! Did you find any relevant documentation or was that something you just figured out digging through the source code?
